repos/fdbt-site/src/pages/api/editFareStageMatching.ts

```diff
@@ -0,0 +1,146 @@
+import { NextApiResponse } from 'next';
+import {
+    getFareZonesEditTicket,
+    getMatchingFareZonesAndUnassignedStopsFromForm,
+    isFareStageUnassignedEditTicket,
+} from '../../utils/apiUtils/matching';
+import {
+    MATCHING_JSON_ATTRIBUTE,
+    MATCHING_JSON_META_DATA_ATTRIBUTE,
+    EDIT_FARE_STAGE_MATCHING_ATTRIBUTE,
+    DIRECTION_ATTRIBUTE,
+} from '../../constants/attributes';
+import { ErrorInfo, NextApiRequestWithSession } from '../../interfaces';
+import { redirectTo, redirectToError } from '../../utils/apiUtils';
+import { putUserDataInProductsBucketWithFilePath } from '../../utils/apiUtils/userData';
+import { getSessionAttribute, updateSessionAttribute } from '../../utils/sessions';
+import { getFareStagesFromTicket } from '../editFareStageMatching';
+import { ReturnTicket, SingleTicket, WithIds } from '../../interfaces/matchingJsonTypes';
+import { isReturnTicket } from '../../utils';
+import { MatchingFareZonesData } from '../../interfaces/matchingInterface';
+
+const getSelectedFareStages = (matchingFareZones: MatchingFareZonesData[]): { [key: string]: string } => {
+    const obj: { [key: string]: string } = {};
+
+    matchingFareZones.forEach((fareZone) => {
+        fareZone.stops.forEach((stop) => {
+            obj[stop.atcoCode] = fareZone.name;
+        });
+    });
+    return obj;
+};
+
+export default async (req: NextApiRequestWithSession, res: NextApiResponse): Promise<void> => {
+    try {
+        const ticket = getSessionAttribute(req, MATCHING_JSON_ATTRIBUTE) as
+            | WithIds<SingleTicket>
+            | WithIds<ReturnTicket>
+            | undefined;
+        const ticketMetaData = getSessionAttribute(req, MATCHING_JSON_META_DATA_ATTRIBUTE);
+        const { overrideWarning } = req.body;
+
+        // edit mode
+        if (!ticket || !ticketMetaData) {
+            throw new Error('Ticket details not found');
+        }
+
+        if (!(ticket.type === 'single' || isReturnTicket(ticket))) {
```

Comment:
we can combine this into the check above, like this:

` if (!ticket || !ticketMetaData || !(ticket.type === 'single || isReturnTicket(ticket)) {`

Reply:
And then just change the error message to say 'Ticket invalid or not found'.